Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM ubuntu:trusty
+MAINTAINER Boik boik@tdohacker.org
+
+ENV HOME /root
+
+ARG DOMAIN
+ARG EMAIL
+ARG MODE=dev
+ARG TYPE=self
+ARG KEY
+ARG CRT
+
+WORKDIR $HOME
+RUN apt-get update \
+  && apt-get install -y rsync ca-certificates
+ADD . $HOME
+RUN chown www-data:www-data $HOME
+RUN ./extra/provision.sh -m $MODE -c $TYPE -k $KEY -C $CRT -D $DOMAIN -e $EMAIL -s `pwd` --docker
+RUN apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```

Comment:
I'm not docker expert, but I've heard it's better practice to put lines like this in the same `RUN` command as the ones where the files were added (the `apt-get update` command). The reason is even though you delete them before the final image is created, docker has already created a FS layer that includes the files, so the image size still increases.